sdk/daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/v2/ledgerinteraction/grpcLedgerClient/GrpcLedgerClient.scala

```diff
@@ -79,10 +79,13 @@ class GrpcLedgerClient(
       explicitPackageId: Boolean,
   ): api.Identifier = {
     val converted = toApiIdentifier(identifier)
-    def pkgName = Runner
-      .getPackageName(compiledPackages, identifier.packageId)
-      .getOrElse(throw new IllegalArgumentException("Couldn't get package name"))
-    if (explicitPackageId) converted else converted.copy(packageId = "#" + pkgName)
+
+    // Package name will not be returned for non-upgradable packages
+    Runner
+      .getUpgradablePackageName(compiledPackages, identifier.packageId)
```

Comment:
I'm guessing the reason `getPackageName` is defined in `Runner` initially is because it is also used in `runWithClients`. But I'm not sure `getUpgradablePackageName` has to be: it could be defined in `GrpcClient` or even inlined here.

The reason I'm mentioning this is because I find that returning the package name but also returning None if the package is not upgradable seems to be mixing concerns a little bit. On the other hand, you don't want to look up the signature twice: once for the name, once for checking upgradability, so I can see why they're merged. But maybe it'd be ok to simply inline that logic in `toApiIdentifierUpgrades`?

Not feeling very strongly about it, your call.

Reply:
Scala put up a bit of a fight but have done.